client/components/dot-pager/index.jsx

```diff
@@ -86,18 +85,27 @@ export const DotPager = ( {
 	const [ currentPage, setCurrentPage ] = useState( 0 );
 	const [ pagesStyle, setPagesStyle ] = useState();
 	const pagesRef = useRef();
-	const [ resizeObserver, sizes ] = useResizeObserver();
 	const numPages = Children.count( children );
-
-	useEffect( () => {
+	const updateLayout = useCallback( () => {
 		if ( ! hasDynamicHeight ) {
 			return;
 		}
 
-		const targetHeight = pagesRef.current?.children[ currentPage ]?.offsetHeight;
-
+		const targetHeight = pagesRef.current?.querySelector( '.is-current' )?.offsetHeight;
```

Comment:
Avoid `currentPage` to create a new `updateLayout` function. Another way is `useRef` to get the latest `currentPage`. I'm okay with both as I think using `querySelector` won't have any performance issues for this case.